Tests/MongoSwiftTests/MongoCollectionTests.swift

```diff
@@ -320,4 +325,76 @@ final class MongoCollectionTests: XCTestCase {
         let docHint = AggregateOptions(hint: .indexSpec(["hi": 1]))
         expect(try encoder.encode(docHint)).to(equal(["hint": ["hi": 1] as Document]))
     }
+
+    func testFindOneAndDelete() throws {
+        // test using maxTimeMS
+        let opts1 = FindOneAndDeleteOptions(maxTimeMS: 100)
+        let result1 = try self.coll.findOneAndDelete(["cat": "cat"], options: opts1)
+        expect(result1).to(equal(self.doc2))
+        expect(try self.coll.count()).to(equal(1))
+
+        // test using a write concern
+        let opts2 = FindOneAndDeleteOptions(writeConcern: try WriteConcern(w: .majority))
+        let result2 = try self.coll.findOneAndDelete([:], options: opts2)
+        expect(result2).to(equal(self.doc1))
+        expect(try self.coll.count()).to(equal(0))
+
+        // test invalid maxTimeMS throws error
+        let invalidOpts1 = FindOneAndDeleteOptions(maxTimeMS: 0)
+        let invalidOpts2 = FindOneAndDeleteOptions(maxTimeMS: -1)
```

Comment:
Would it make more sense to validate options when they're first constructed, or is that just be annoying for users since they'd have to `try` both this initialization and the actual executing method. I assume the convention is that this is a simple value container with no rules on its own.